src/Raven.Server/Smuggler/Documents/DatabaseSmuggler.cs

```diff
@@ -351,6 +338,78 @@ private SmugglerProgressBase.Counts ProcessIndexes(SmugglerResult result)
+                            "For example: " + Environment.NewLine +
+                                "\tfrom doc in doc.Orders" + Environment.NewLine +
+                                "\tlet company = LoadDocument(doc.Company, \"Companies\")" + Environment.NewLine +
+                                "\tselect new {" + Environment.NewLine +
+                                    "\t\tCompanyName: comapny.Name" + Environment.NewLine +
+                                "\t}" + Environment.NewLine;
+                }
+                else if (retryOnError && 
+                         (exceptionMessage.Contains("The name 'AbstractIndexCreationTask' does not exist in the current context") ||
+                          exceptionMessage.Contains("The name 'SpatialIndex' does not exist in the current context") ||
+                          exceptionMessage.Contains("The type or namespace name 'Abstractions' does not exist in the namespace 'Raven' (are you missing an assembly reference?)")))
+                {
+                    var maps = new HashSet<string>();
+                    foreach (var map in indexDefinition.Maps)
+                    {
+                        maps.Add(ReplaceLegacyMethods(map));
+                    }
+                    indexDefinition.Maps = maps;
+
+                    if (indexDefinition.Reduce != null)
+                    {
+                        indexDefinition.Reduce = ReplaceLegacyMethods(indexDefinition.Reduce);
+                    }
+
+                    result.AddMessage($"Replaced legacy methods in index: {indexDefinition.Name}: " + Environment.NewLine +
+                                      "AbstractIndexCreationTask.SpatialGenerate -> CreateSpatialField" + Environment.NewLine +
+                                      "SpatialIndex.Generate -> CreateSpatialField" + Environment.NewLine +
+                                      "new Raven.Abstractions.Linq.DynamicList was removed");
+
+                    WriteIndex(result, indexDefinition, actions, retryOnError: false);
+                    return;
+                }
+
+                result.Indexes.ErroredCount++;
+                var errorMessage = $"Could not write index '{indexDefinition.Name}', error: {exceptionMessage}" + Environment.NewLine +
+                                   $"Maps: [{Environment.NewLine}{string.Join($", {Environment.NewLine}", indexDefinition.Maps)}{Environment.NewLine}]";
+
+                if (indexDefinition.Reduce != null)
+                {
+                    errorMessage += Environment.NewLine + $"Reduce: {indexDefinition.Reduce}";
+                }
+                result.AddError(errorMessage);
+            }
+
+            string ReplaceLegacyMethods(string str)
+            {
+                return str
+                    .Replace("AbstractIndexCreationTask.SpatialGenerate", "CreateSpatialField")
```

Comment:
We do not want to replace them, just write a proper error as we do for LoadDocument

Reply:
fixed